core/commands/storage/upload/upload/dc_repair_router.go

```diff
@@ -256,34 +263,37 @@ func doRepair(ctxParams *uh.ContextParams, params *RepairContractParams) error {
 			go checkPaymentFromClient(ctxParams, paidIn, signedContractID)
 			paid := <-paidIn
 			if !paid {
-				log.Error("contract is not paid", zap.String("contractId", repairContract.RepairContractId))
+				logger.Debugf("contract is not paid", zap.String("contractId", repairContract.RepairContractId))
+				return fmt.Errorf("contract is not paid: %s", repairContract.RepairContractId)
 			}
-			_, err = downloadAndRebuildFile(ctxParams, repairContract.FileHash, repairContract.LostShardHash)
+			lostShards := strings.Join(repairContract.LostShardHash, ",")
+			err = cmdenv.DownloadAndRebuildFile(ctxParams.Req, res, ctxParams.Api, repairContract.FileHash, lostShards)
 			if err != nil {
-				log.Errorf("download and rebuild file error: [%v]", err)
+				logger.Debugf("download and rebuild file error", zap.String("lostShards", lostShards))
 				return err
 			}
 			err = challengeLostShards(ctxParams, repairContract)
 			if err != nil {
-				log.Errorf("challenge lost shards error: [%v]", err)
+				logger.Errorf("challenge lost shards error: [%v]", err)
 				return err
 			}
 			fileStatus, err := uploadShards(ctxParams, repairContract)
 			if err != nil {
-				log.Errorf("repairer uploads shards error: [%v]", err)
+				logger.Errorf("repairer uploads shards error: [%v]", err)
 				return err
 			}
 			if fileStatus == nil {
+				cmdenv.RmPin(ctxParams.Req, res, ctxParams.N, ctxParams.Api, ctxParams.Cfg, arguments, true, false)
 				return nil
 			}
 			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
 			defer cancel()
 			err = submitFileStatus(ctx, ctxParams.Cfg, fileStatus)
 			if err != nil {
-				log.Errorf("submit file store status error: [%v]", err)
+				logger.Errorf("submit file store status error: [%v]", err)
 				return err
 			}
-			unpinLocalStorage(ctxParams, repairContract.FileHash)
+			cmdenv.RmPin(ctxParams.Req, res, ctxParams.N, ctxParams.Api, ctxParams.Cfg, arguments, true, false)
```

Comment:
Difference between these two?

Reply:
unpinLocalStorage is the function to call the pin rm command by http, it has commented and replaced by cmdenv.RmPin.